base/collection.go

```diff
@@ -183,6 +189,7 @@ type GocbV2Bucket struct {
 	queryOps                                             chan struct{} // Manages max concurrent query ops
 	kvOps                                                chan struct{} // Manages max concurrent kv ops
 	clusterCompatMajorVersion, clusterCompatMinorVersion uint64        // E.g: 6 and 0 for 6.0.3
+	ClusterUUID                                          string        // uuid of cluster
```

Comment:
What's the benefit of defining ClusterUUID on GocbV2Bucket, instead using the version we already have on DatabaseContext (the lazy loading done in GetServerUUID())

Reply:
The benefit here was that I would do it directly after making a call for a connection to a bucket so it would be less inclined toward failure because you just wouldn't a bucket if you couldn't immediately make a gocb connection and hit this end point.

Reply:
Moved to NewDatabaseContext with a somewhat arbitrary timeout, used elsewhere in the file, but it's actually used for bootstrap connection and not the NewDatabaseContext. If you have better ideas on numbers to choose, I'm all ears.